MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,72 +1,165 @@
+            correctAnswer: true),
+        QuizQuestion(
+            image: "The Green Knight",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: true),
+        QuizQuestion(
+            image: "Old",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: false),
+        QuizQuestion(
+            image: "The Ice Age Adventures of Buck Wild",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: false),
+        QuizQuestion(
+            image: "Tesla",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: false),
+        QuizQuestion(
+            image: "Vivarium",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: false)
+    ]
+    
+    private var currentQuestionIndex = 0
+    private var correctAnswers = 0
+    
     // MARK: - Lifecycle
     override func viewDidLoad() {
         super.viewDidLoad()
+        show(quizStep: convert(model: questions[currentQuestionIndex]))
+        imageView.layer.masksToBounds = true
+        imageView.layer.cornerRadius = 20
+    }
+    
+    struct QuizStepViewModel {
+        let image: UIImage
+        let question: String
+        let questionNumber: String
+    }
+    
+    struct QuizResultsViewModel {
+        let title: String
+        let text: String
+        let buttonText: String
+    }
+    
+    struct QuizQuestion {
+        let image: String
+        let text: String
+        let correctAnswer: Bool
+    }
+    
+    @IBAction private func noButtonClicked(_ sender: UIButton) {
+        let answerToQuestion = false
+        let currentQuestion = questions[currentQuestionIndex]
+        showAnswerResult(isCorrect: answerToQuestion == currentQuestion.correctAnswer)
+    }
+    
+    @IBAction private func yesButtonClicked(_ sender: UIButton) {
+        let answerToQuestion = true
+        let currentQuestion = questions[currentQuestionIndex]
+        showAnswerResult(isCorrect: answerToQuestion == currentQuestion.correctAnswer)
+    }
+    
+    //MARK: - Methods
+    private func show(quizStep: QuizStepViewModel) {
+        imageView.image = quizStep.image
+        textLabel.text = quizStep.question
+        counterLabel.text = quizStep.questionNumber
+    }
+    
+    private func show(quizResult: QuizResultsViewModel) {
+        let alert = UIAlertController(
+            title: quizResult.title,
+            message: quizResult.text,
+            preferredStyle: .alert)
+        let action = UIAlertAction(
+            title: quizResult.buttonText,
+            style: .default) { _ in
+                self.currentQuestionIndex = 0
+                self.correctAnswers = 0
+                let firstQuestion = self.questions[self.currentQuestionIndex]
+                let viewModel = self.convert(model: firstQuestion)
+                self.show(quizStep: viewModel)
+            }
```

Comment:
В будущих спринтах вам будут рассказывать про слабый список захвата. Здесь используется замыкание, которое на данный момент сильно захватывает ссылку на класс. Такой код может вызывать утечки памяти. Без подробностей, по желанию, предлагаю добавить `[weak self]`. Такая добавка потребует от вас вызвать методы и свойства класса через `self?.` Поэтому внутренности предлагаю вынести в отдельную функцию. Например, так:

```suggestion
            style: .default) { [weak self] _ in
                self?.restartQuiz()
            }
```

Reply:
Буду знать, спасибо)